way_to_home/custom_user/views.py

```diff
@@ -63,3 +66,37 @@ def log_in(request):
         return HttpResponse('invalid credentials', status=400)
     login(request, user=user)
     return HttpResponse('operation was successful provided', status=200)
+
+
+@require_http_methods(["GET"])
+def auth_google(request):
+    """Function that provides getting url for confirmation access to user's data."""
+    google = OAuth2Session(client_id=CLIENT_ID, redirect_uri=REDIRECT_URL, state=STATE, scope=SCOPE)
+    data = google.authorization_url(url=AUTH_URI, state=STATE)[0]
+    if data:
+        return redirect(data)
+    return HttpResponse(status=400)
+
+
+@require_http_methods(["GET"])
+def signin_google(request):
+    """Function that provides user registration or authorization via google."""
+    google = OAuth2Session(client_id=CLIENT_ID, redirect_uri=REDIRECT_URL, state=STATE, scope=SCOPE)
+    try:
+        code = request.GET["code"]
+    except ValueError:
+        return HttpResponse(status=400)
```

Comment:
`return HttpResponse("Failed", status=400)`
return response with message that informs client what`s happend